# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1385295096 -3600
#      Sun Nov 24 13:11:36 2013 +0100
# Node ID ea1db0ede4ce0a0ddcb628dc28e4af00784c4a21
# Parent  a99b7d656a6cb6092720f7b28a4d82d000bf7c0a
make sure Octave is displayed in a global menu bar, not another widget name

* main-window.cc (set_window_layout): show main window first but minimized
  in order to avoid flickering

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -904,19 +904,22 @@ main_window::set_window_layout (QSetting
           else
             {
               widget->make_widget ();
               widget->setVisible (visible);       // not floating -> show
             }
         }
     }
 
+  // show main first but minimized to avoid flickering
+  // otherwise the name of a floating widget is shown in a global menu bar
+  showMinimized ();
   restoreState (settings->value ("MainWindow/windowState").toByteArray ());
   restoreGeometry (settings->value ("MainWindow/geometry").toByteArray ());
-  show ();  // main window is ready and can be shown (as first window)
+  show ();
 
   // show floating widgets after main win to ensure "Octave" in central menu
   foreach (octave_dock_widget *widget, float_and_visible)
     {
 #if ! defined (Q_OS_WIN32)
       widget->make_window ();
 #endif
       if (settings->value ("DockWidgets/" + widget->objectName () + "_minimized").toBool ())
